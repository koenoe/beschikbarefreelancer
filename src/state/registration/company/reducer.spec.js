// @flow
import companyReducer from 'state/registration/company/reducer';

import type { State } from 'state/registration/company/reducer';

describe('Registration reducer: company', () => {
  it('should set `isLoading` to true when `REGISTER_COMPANY_REQUEST`', () => {
    const beforeState: State = {
      isLoading: false,
    };
    const afterState: State = companyReducer(beforeState, {
      type: 'registration/COMPANY_REQUEST',
      email: 'test@koenromers.com',
    });
    expect(afterState).toMatchSnapshot();
    expect(afterState).not.toBe(beforeState);
  });

  it('should set `success` to true when `registration/COMPANY_SUCCESS`', () => {
    const beforeState: State = {
      isLoading: true,
      success: false,
    };
    const afterState: State = companyReducer(beforeState, {
      type: 'registration/COMPANY_SUCCESS',
      success: true,
    });
    expect(afterState).toMatchSnapshot();
    expect(afterState).not.toBe(beforeState);
  });

  it('should have an `error` when `registration/COMPANY_FAILURE`', () => {
    const beforeState: State = {
      isLoading: true,
      error: null,
    };
    const afterState: State = companyReducer(beforeState, {
      type: 'registration/COMPANY_FAILURE',
      error: new Error('fail whale'),
    });
    expect(afterState).toMatchSnapshot();
    expect(afterState).not.toBe(beforeState);
  });
});
