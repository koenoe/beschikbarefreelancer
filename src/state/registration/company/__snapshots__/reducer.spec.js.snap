// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration reducer: company should have an \`error\` when \`registration/COMPANY_FAILURE\` 1`] = `
Object {
  "error": [Error: fail whale],
  "isLoading": false,
}
`;

exports[`Registration reducer: company should set \`isLoading\` to true when \`REGISTER_COMPANY_REQUEST\` 1`] = `
Object {
  "isLoading": true,
}
`;

exports[`Registration reducer: company should set \`success\` to true when \`registration/COMPANY_SUCCESS\` 1`] = `
Object {
  "isLoading": false,
  "success": true,
}
`;
